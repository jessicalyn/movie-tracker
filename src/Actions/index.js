export const addMovies = (movies) => ({
  type: "ADD_MOVIES",
  movies
})

export const addUser = (user) => ({
  type: "ADD_USER",
  user
  
})

export const loginUser = (email) => ({
  type: "LOGIN_USER",
  email
})

