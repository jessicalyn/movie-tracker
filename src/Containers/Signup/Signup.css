.signup{
    background: url("../../images/popcorn.jpg") no-repeat;
    background-size:cover;
    height:700px;
     display: flex;
    justify-content: center;
    align-items: center;

}
.signup-box{
    background-color:white;
    width:30%;
    height: 500px;
    padding: 20px; 
    border: 10px solid black;
}

.user-name img {
    width: 45px;
    height: 48px;
}

.user-name{
    background-color: #cccc;
    display: flex;
    padding: 10px;
    background-color: white;
    margin-bottom: 22px;
    justify-content: space-between;
}

.input-box{
    display: flex;
    padding: 10px;
    background-color: white;
    margin-bottom: 22px;
    -webkit-justify-content: space-between;
    justify-content: space-between;
}
form button {
    background-color: #e50914;
}

.signup-btn{
    background-color: #e50914;
    color:white;
    border:1px solid white;
    width:100%;
}