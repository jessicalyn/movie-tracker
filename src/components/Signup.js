import React from "react";
import { Component } from "react"

export class Signup extends Component {
    constructor(){
        super();
        this.state={
            name: "",
            email: "",
            password: "",
            error: ""
        }
    }
    handleChange = (event) => {
        const { name, value } = event.target
        this.setState({
            [name]: value
        })
    }

    handleSubmit = async (event) => {
        event.preventDefault()
        const url = "http://localhost:3000/api/users/new"
        const userInfo = {
            name: this.state.name,
            email: this.state.email,
            password: this.state.password
        }
        fetch(url,{
            method: "POST",
            body: JSON.stringify(userInfo),
            headers: {
              "Content-Type": "application/json"
            }
          })
            .then(response => response.json())
            .then(response => response.JSON.stringify(response))
            .then(error => this.setState({error: error.message}))

    }

    render(){
        return (
            <section>
        <form onSubmit={this.handleSubmit}>
          <input
            type="text"
            name="name"
            placeholder="name"
            value= {this.state.name}
            onChange={this.handleChange}
            
            
          />
          <input
            type="text"
            name="email"
            placeholder="email"
            value= {this.state.email}
            onChange={this.handleChange}
            
            
          />
          <input
            type="text"
            name="password"
            placeholder="password"
            value={this.state.password}
            onChange={this.handleChange}
            
            
          />
          

          <button>Sign up</button>
          
        </form>
      </section>
        )
    }
}

