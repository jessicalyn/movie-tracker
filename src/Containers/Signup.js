import React from "react";
import { Component } from "react"
import { connect } from "react-redux"
import { loginUser } from "../Actions/index"
import { Route, Redirect } from 'react-router'
import  user  from '../images/user.png'
import email from '../images/email-icon.png'
import password from '../images/pwd.png'
import './Movies/Signup.css'

export class Signup extends Component {
    constructor(){
        super();
        this.state={
            name: "",
            email: "",
            password: "",
            error: ""
        }
    }

    handleChange = (event) => {
        const { name, value } = event.target
        this.setState({ [name]: value })
    }

    handleSubmit = async (event) => {
      event.preventDefault()
      const url = "http://localhost:3000/api/users/new"
      const userInfo = {
        name: this.state.name,
        email: this.state.email,
        password: this.state.password
      }
       const response = await fetch(url, {
          method: 'POST',
          body: JSON.stringify(userInfo),
          headers:{
            'Content-Type': 'application/json'
          }
        })
        const result = await response.json()
        console.log(result, "response")
        if(result.status === "success"){
         return this.props.loginUser(result.id)
        }else {
          this.setState({
            error: "Email has already been used, please Login."
          })
        }
    }

    render(){
      return (
        <section className="signup">
        <div className="signup-box">
        <form onSubmit={this.handleSubmit}>
          <div className="signup-header">
              <h3>One step and you're done!
              We hate paperwork, too.</h3>
          </div>
              <div className="user-name input-box">
                <img alt="user-icon" src={user}></img>
                <input
                  type="text"
                  name="name"
                  placeholder="name"
                  value= {this.state.name}
                  onChange={this.handleChange}
            />
            </div>
          <div className="email input-box">
          <img alt="user-icon" src={email}></img>
              <input
                type="text"
                name="email"
                placeholder="email"
                value= {this.state.email}
                onChange={this.handleChange}
          />
          </div>
          <div className="password input-box">
          <img alt="user-icon" src={password}></img>
          <input
            type="text"
            name="password"
            placeholder="password"
            value={this.state.password}
            onChange={this.handleChange}
          />
          </div>
          <button class="signup-btn">Sign up</button>
        </form>
        </div>
        {this.state.error && this.state.error}
        <Route exact path ='/Signup' render={()=> (
          this.props.user.id && <Redirect to="/"/>
        )}/>
      </section>
        )
    }
}
export const mapStateToProps = (state) => ({
  user: state.user
})

export const mapDispatchToProps = (dispatch) => ({
    loginUser: (user) => dispatch(loginUser(user))
  })

export default connect(mapStateToProps, mapDispatchToProps)(Signup)
