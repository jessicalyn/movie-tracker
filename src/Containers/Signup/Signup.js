import React from "react";
import { Component } from "react"
import { connect } from "react-redux"
import { loginUser, hasError } from "../../Actions/index"
import { Route, Redirect } from 'react-router'
import { fetchData } from '../../Utils/fetchData'
import { fetchOptionsCreator } from '../../Utils/fetchOptionsCreator'
import './Signup.css'

export class Signup extends Component {
    constructor(){
        super();
        this.state = {
            name: "",
            email: "",
            password: ""
        }
    }

    handleChange = (event) => {
        const { name, value } = event.target
        this.setState({ [name]: value })
    }

    handleSubmit = async (event) => {
      event.preventDefault()
      const url = "http://localhost:3000/api/users/new"
      const body = {
        name: this.state.name,
        email: this.state.email,
        password: this.state.password
      }
      try {
      const options = await fetchOptionsCreator('POST', body)
      const result = await fetchData(url, options)
        if(result.status === "success"){
         return this.props.loginUser(result.id)
        }
       } catch(error) {
        const message = "Email has already been used, please Login."
        return this.props.hasError(message)
        }
    }

    render(){
      return (
        <section>
        <form onSubmit={this.handleSubmit}>
          <input
            type="text"
            name="name"
            placeholder="name"
            value= {this.state.name}
            onChange={this.handleChange}
          />
          <input
            type="text"
            name="email"
            placeholder="email"
            value= {this.state.email}
            onChange={this.handleChange}
          />
          <input
            type="text"
            name="password"
            placeholder="password"
            value={this.state.password}
            onChange={this.handleChange}
          />
          <button className="signup-button">Sign up</button>
        </form>
        {this.props.error && this.props.error}
        <Route exact path ='/Signup' render={()=> (
          this.props.user.id && <Redirect to="/"/>
        )}/>
        </section>
      )
    }
}
export const mapStateToProps = (state) => ({
  user: state.user,
  error: state.error
})

export const mapDispatchToProps = (dispatch) => ({
    loginUser: (userId) => dispatch(loginUser(userId)),
    hasError: (message) => dispatch(hasError(message))
  })

export default connect(mapStateToProps, mapDispatchToProps)(Signup)
