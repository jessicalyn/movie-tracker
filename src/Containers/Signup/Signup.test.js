import React from 'react'
import { Signup, mapStateToProps, mapDispatchToProps } from './Signup'
import { shallow } from 'enzyme'
import { updateUser } from '../../Actions'
import { Route, Redirect } from 'react-router'

describe('Signup', () => {
  describe('Signup', () => {
    let wrapper

    beforeEach(() => {
      const mockUser = { id: 1, name: "Taylor" }
      wrapper = shallow(
        <Signup 
          loginUser={ jest.fn() } 
          user={ mockUser } 
        />
      )
    })

    it('should match the snapshot with all data passed in', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should have the proper default state', () => {
      const mockState = {
        name: "",
        email: "",
        password: ""
      }
      expect(wrapper.state()).toEqual(mockState)
    })

    it('should update state when handleChange is called', () => {
      const mockEvent = {
        target: {
          name: "email",
          value: "archie@gmail.com"
        }
      }
      expect(wrapper.state('email')).toEqual("")
      wrapper.instance().handleChange(mockEvent)
      expect(wrapper.state('email')).toEqual("archie@gmail.com")
    })

    it('should invoke handleSubmit when form button is clicked', () => {
      const mockHandleSubmit = jest.fn()
      wrapper.find('.signup-button').simulate('click')
      expect(wrapper.instance().mockHandleSubmit()).toBeCalled()
    })

  })

    describe('mapStateToProps', () => {
      it('should return an object with the current user', () => {
        const mockState = {
          movies: [
            {title: "Cars", poster_path: "car.jpg"},
            {title: "How To Train Your Dragon", poster_path: "dragon.jpg"}
          ],
          user: {
            id: 1,
            name: "Taylor"
          }
        }
        const expected = { user: { id: 1, name: "Taylor"} }

        const mappedProps = mapStateToProps(mockState)

        expect(mappedProps).toEqual(expected)
      })
    })

    describe('mapDispatchToProps', () => {
      it('should call dispatch when using updateUser within mapDispatchToProps', () => {
        const mockDispatch = jest.fn()
        const actionToDispatch = updateUser(1)
        const mappedProps = mapDispatchToProps(mockDispatch)

        mappedProps.updateUser(1)

        expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
      })
    })
})