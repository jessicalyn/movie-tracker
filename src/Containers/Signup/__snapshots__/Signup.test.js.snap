// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Signup should match the snapshot with all data passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Signup
    loginUser={[MockFunction]}
    user={
      Object {
        "id": 1,
        "name": "Taylor",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          onSubmit={[Function]}
        >
          <input
            name="name"
            onChange={[Function]}
            placeholder="name"
            type="text"
            value=""
          />
          <input
            name="email"
            onChange={[Function]}
            placeholder="email"
            type="text"
            value=""
          />
          <input
            name="password"
            onChange={[Function]}
            placeholder="password"
            type="text"
            value=""
          />
          <button>
            Sign up
          </button>
        </form>,
        "",
        <Route
          exact={true}
          path="/Signup"
          render={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="name"
              onChange={[Function]}
              placeholder="name"
              type="text"
              value=""
            />,
            <input
              name="email"
              onChange={[Function]}
              placeholder="email"
              type="text"
              value=""
            />,
            <input
              name="password"
              onChange={[Function]}
              placeholder="password"
              type="text"
              value=""
            />,
            <button>
              Sign up
            </button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "name",
              "onChange": [Function],
              "placeholder": "name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "email",
              "onChange": [Function],
              "placeholder": "email",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "password",
              "onChange": [Function],
              "placeholder": "password",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign up",
            },
            "ref": null,
            "rendered": "Sign up",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/Signup",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            onSubmit={[Function]}
          >
            <input
              name="name"
              onChange={[Function]}
              placeholder="name"
              type="text"
              value=""
            />
            <input
              name="email"
              onChange={[Function]}
              placeholder="email"
              type="text"
              value=""
            />
            <input
              name="password"
              onChange={[Function]}
              placeholder="password"
              type="text"
              value=""
            />
            <button>
              Sign up
            </button>
          </form>,
          "",
          <Route
            exact={true}
            path="/Signup"
            render={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="name"
                onChange={[Function]}
                placeholder="name"
                type="text"
                value=""
              />,
              <input
                name="email"
                onChange={[Function]}
                placeholder="email"
                type="text"
                value=""
              />,
              <input
                name="password"
                onChange={[Function]}
                placeholder="password"
                type="text"
                value=""
              />,
              <button>
                Sign up
              </button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "name",
                "onChange": [Function],
                "placeholder": "name",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "email",
                "onChange": [Function],
                "placeholder": "email",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "password",
                "onChange": [Function],
                "placeholder": "password",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign up",
              },
              "ref": null,
              "rendered": "Sign up",
              "type": "button",
            },
          ],
          "type": "form",
        },
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/Signup",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
